controllers/imagerepository_controller.go

```diff
@@ -210,6 +265,28 @@ func (r *ImageRepositoryReconciler) scan(ctx context.Context, imageRepo *imagev1
 		options = append(options, remote.WithAuth(auth))
 	}
 
+	if accountId, awsEcrRegion, _, _, _, err := parseAwsImageURL(imageRepo.Spec.Image); err == nil {
+		if _, present := os.LookupEnv("USE_ECR"); present {
```

Comment:
I would prefer this as a flag to the controller, rather than an environment variable. Really only because that's more in the spirit of controller-runtime with its config file support -- in most respects environment vars and flags work equally well.

I think it _is_ appropriate for it to be set at the controller level, since it's up to the platform admin to grant global access to ambient authority, rather than for users to claim it. (If there is a reason to opt in per image repository, then fine, but I would still want the flag on the controller to explicitly grant access to everyone who wants it).

Reply:
Yes, I'll try to find out how to access the command line and change the PR.... 

But I have no Idea what "instead of merging main into this branch, please rebase this branch on main" means...

Reply:
@squaremo  I guess I got the command-line option working. As for the rebase topic, I brushed up on Git theory, but I have no clue on how to execute it here on Github without damaging anything (Yes, it's my very first Github PR).

Reply:
> I have no clue on how to execute it here on Github without damaging anything (Yes, it's my very first Github PR).

First GitHub PR high five! ✋ 
I don't think GitHub has an in-app way to do this; it's something you would do on your local computer, after cloning your fork of this repo.

> But I have no Idea what "instead of merging main into this branch, please rebase this branch on main" means...

Since you're working from a fork, you'll have to add this repo as a `<remote>`, first:

    git remote add upstream https://github.com/fluxcd/image-reflector-controller

Also, it's worth working in a branch, and reserving `main` for keeping track of the upstream main branch:

    git switch -c ecr-auto-creds
    git push -u origin ecr-auto-creds

(The second command pushes the new branch to your forked repo, and `-u` sets it as the default place to push any further commits on that branch)

Then, rebasing on the main branch here can be done with

    git pull upstream main --rebase

A lot of the time, this will succeed without any more work from you. If you get into trouble, you can `git rebase --abort` to take you back to the state before the rebase (or `pull --rebase`).

To push the branch back to GitHub, you'll need to use `--force-with-lease`, because you've replaced the commits in the branch:

    git push --force-with-lease # it knows where to push to, because of `-u` above

Reply:
@squaremo Thanks! I followed your instructions, and all commands succeeded. But apparently it is not possible to change the source branch of a Github PR, so it's still from my "main" branch :(